layers/+lang/javascript/packages.el

```diff
@@ -78,8 +75,7 @@
     :mode "\\.m?js\\'"
     :init
     (progn
-      (add-hook 'js2-mode-local-vars-hook
-                #'spacemacs//javascript-setup-backend)
+      (spacemacs//javascript-setup-backend)
```

Comment:
You cannot do that, the value of  `javascript-backend` won't be read from `.dir-locals.el` files preventing users to be able to choose the backend on a per-project basis, see: https://github.com/syl20bnr/spacemacs/blob/develop/layers/+lang/javascript/README.org#L59
The functions hooked to `xxx-mode-local-vars-hook` are read after emacs read the local directory files. 
See https://github.com/syl20bnr/spacemacs/blob/develop/layers/+spacemacs/spacemacs-defaults/config.el#L205-L207 and Emacs documentation for more info.

Why did you remove this function from `js2-mode-local-vars-hook` exactly ?

Reply:
It is because I made some change to hook `lsp-javascript-typescript-enable` with `js-mode`. But this is not necessary anymore. I pushed a new commit to fix this.